src/sonic-containercfgd/containercfgd/containercfgd.py

```diff
@@ -0,0 +1,198 @@
+        return object_type
+    return wrapper
+
+
+@config_handler(SYSLOG_CONFIG_FEATURE_TABLE)
+class SyslogHandler:
+    # syslog conf file path in docker
+    SYSLOG_CONF_PATH = '/etc/rsyslog.conf'
+    # temporary syslog conf file path in docker
+    TMP_SYSLOG_CONF_PATH = '/tmp/rsyslog.conf'
+
+    # Regular expressions to extract value from rsyslog.conf
+    INTERVAL_PATTERN = '.*SystemLogRateLimitInterval\s+(\d+).*'
+    BURST_PATTERN = '.*SystemLogRateLimitBurst\s+(\d+).*'
+    TARGET_IP_PATTERN = '.*target="(.*?)".*'
+
+    def __init__(self):
+        self.current_interval, self.current_burst, self.target_ip = self.parse_syslog_conf()
+
+    def handle_config(self, table, key, data):
+        """Handle CONFIG DB change. Callback by ConfigDBConnector.
+
+        Args:
+            table (str): CONFIG DB table name
+            key (str): Key of the changed entry
+            data (dict): Data of the entry: {<field_name>: <field_value>}
+        """
+        try:
+            if key != container_name:
+                return
+            self.update_syslog_config(data)
+        except Exception as e:
+            logger.log_error('Failed to config syslog for container {} with data {} - {}'.format(key, data, e))
+
+    def handle_init_data(self, init_data):
+        """Handle initial data in CONFIG DB. Callback by ConfigDBConnector.
+
+        Args:
+            init_data (dict): Initial data when first time connecting to CONFIG DB. {<table_name>: {<field_name>: <field_value>}}
+        """
+        if SYSLOG_CONFIG_FEATURE_TABLE in init_data:
+            if container_name in init_data[SYSLOG_CONFIG_FEATURE_TABLE]:
+                self.update_syslog_config(init_data[SYSLOG_CONFIG_FEATURE_TABLE][container_name])
+
+    def update_syslog_config(self, data):
+        """Parse existing syslog conf and apply new syslog conf.
+
+        Args:
+            data (dict): Data of the entry: {<field_name>: <field_value>}
+        """
+        new_interval = '0' if not data else data.get(SYSLOG_RATE_LIMIT_INTERVAL, '0')
+        new_burst = '0' if not data else data.get(SYSLOG_RATE_LIMIT_BURST, '0')
+
+        if new_interval == self.current_interval and new_burst == self.current_burst:
+            logger.log_notice('Syslog rate limit configuration does not change, ignore it')
+            return
+
+        logger.log_notice(f'Configure syslog rate limit interval={new_interval}, burst={new_burst}')
+
+        if os.path.exists(self.TMP_SYSLOG_CONF_PATH):
+            os.remove(self.TMP_SYSLOG_CONF_PATH)
+        with open(self.TMP_SYSLOG_CONF_PATH, 'w+') as f:
+            output = run_command('sonic-cfggen -d -t /usr/share/sonic/templates/rsyslog-container.conf.j2 -a "{{\\"target_ip\\": \\"{}\\", \\"container_name\\": \\"{}\\" }}"'.format(self.target_ip, container_name))
+            f.write(output)
+        run_command('cp {} {}'.format(self.TMP_SYSLOG_CONF_PATH, self.SYSLOG_CONF_PATH))
+        run_command("supervisorctl restart rsyslogd")
```

Comment:
Can these be split into individual entries (such as `["cp", self.TMP_SYSLOG_CONF_PATH, self.SYSLOG_CONF_PATH]` and passed directly into `subprocess.check_output` instead of using `shlex.split`? That will prevent code injections from outside sources doing something that shouldn't be done.

Reply:
That's true, because it's split into an array, the first token will always be the command that gets executed. However, if, hypothetically, `self.TMP_SYSLOG_CONF_PATH` was `/tmp/file name`, then `shlex.split` will split `cp /tmp/file name /tmp/file_copy` into `["cp", "/tmp/file", "name", "/tmp/file_copy"]`, which is incorrect.

I think considering we know what these arguments should be, just doing the splitting ourselves would be better. (It would also make the JSON argument a couple lines above this a bit easier to read).

Reply:
fixed